src/transformers/trainer.py

```diff
@@ -2820,11 +2779,15 @@ def save_model(self, output_dir: Optional[str] = None, _internal_call: bool = Fa
             ShardedDDPOption.ZERO_DP_2 in self.args.sharded_ddp
             or ShardedDDPOption.ZERO_DP_3 in self.args.sharded_ddp
             or self.fsdp is not None
+            or getattr(self.accelerator.state, "fsdp_plugin", None) is not None
```

Comment:
Would these now be one in the same? (So we can use only the `fsdp_plugin` check, or `self.fsdp` for everything?)

Reply:
E.g. such as how we have the following below:

```python
if len(self.fsdp) > 0 and not self.fsdp_config["xla"]
```

Reply:
both are needed as FSDP XLA isn't supported in accelerate yet

Reply:
Makes sense, thanks :)